silkworm/rpc/ws/websocket_connection.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+   Copyright 2023 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+//
+// Copyright (c) 2003-2020 Christopher M. Kohlhoff (chris at kohlhoff dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#pragma once
+
+#include <array>
+#include <string>
+
+#include <silkworm/infra/concurrency/task.hpp>
+
+#include <boost/asio/io_context.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/thread_pool.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/websocket.hpp>
+
+#include <silkworm/rpc/commands/rpc_api_table.hpp>
+#include <silkworm/rpc/http/channel.hpp>
+#include <silkworm/rpc/http/request_handler.hpp>
+
+namespace silkworm::rpc::http {
+
+//! Represents a single connection from a client via websocket.
+class WebSocketConnection : public std::enable_shared_from_this<WebSocketConnection>, Channel {
```

Comment:
Rename `WebSocketConnection` as `Connection`

Reply:
ok and removed std::enable_shared_from_this<WebSocketConnection>